boudicca.base/eventcollector-client/src/main/kotlin/base/boudicca/api/eventcollector/webui/WebuiService.kt

```diff
@@ -0,0 +1,195 @@
+                            mapSingleCollectionToFrontend(it, singleCollections[it])
+                        }
+        )
+    }
+
+    private fun mapSingleCollectionToFrontend(it: SingleCollection): Map<String, *> {
+        return mapSingleCollectionToFrontend(it.collectorName, it)
+    }
+
+    private fun mapSingleCollectionToFrontend(name: String, it: SingleCollection?): Map<String, *> {
+        if (it != null) {
+            return mapOf(
+                "id" to it.id.toString(),
+                "name" to HtmlUtils.htmlEscape(it.collectorName),
+                "duration" to formatDuration(it.startTime, it.endTime),
+                "startEndTime" to formatStartEndTime(it.startTime, it.endTime),
+                "errorCount" to it.errorCount.toString(),
+                "warningCount" to it.warningCount.toString(),
+                "hasErrors" to (it.errorCount > 0),
+                "hasWarnings" to (it.warningCount > 0),
+                "totalEventsCollected" to (it.totalEventsCollected).toString(),
+            )
+        } else {
+            return mapOf(
+                "id" to null,
+                "name" to name,
+                "duration" to "-",
+                "startEndTime" to "-",
+                "errorCount" to "-",
+                "warningCount" to "-",
+                "hasErrors" to false,
+                "hasWarnings" to false,
+                "totalEventsCollected" to "-",
+            )
+        }
+    }
+
+    private fun mapHttpCallToFrontend(httpCall: HttpCall): Map<String, String> {
+        return mapOf(
+            "url" to httpCall.url!!,
+            "responseCode" to if (httpCall.responseCode == 0) "-" else httpCall.responseCode.toString(),
+            "duration" to formatDuration(httpCall.startTime, httpCall.endTime),
+            "startEndTime" to formatStartEndTime(httpCall.startTime, httpCall.endTime),
+            "postData" to (httpCall.postData ?: ""),
+        )
+    }
+
+    private fun findSingleCollection(id: UUID): SingleCollection? {
+        return Collections.getAllPastCollections()
+            .union(listOfNotNull(Collections.getCurrentFullCollection()))
+            .flatMap { it.singleCollections }
+            .find { it.id == id }
+    }
+
+    private fun findFullCollection(id: UUID): FullCollection? {
+        return Collections.getAllPastCollections().find { it.id == id }
+    }
+
+    private fun parseId(query: String?): UUID? {
+        if (query == null) {
+            return null
+        }
+
+        val id = query.split('&')
+            .map {
+                it.split('=')
+            }
+            .find {
+                it[0] == "id"
+            }
+        if (id != null) {
+            return UUID.fromString(id[1])
+        }
+        return null
+    }
+
+    private fun formatStartEndTime(startTimeInMillis: Long, endTimeInMillis: Long): String {
+        val startTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(startTimeInMillis), ZoneId.of("CET"))
```

Comment:
Hardcoded timezone? could we pull it out to properties as to make it at least configurable when starting the container?